File: apps/site/snippets/voice/overview.swift
import SwiftUI
import Sinua
import SinuaVoice

struct TalkingOrb: View {
    // Any VoiceSource: the mic here, or a vendor source (see the vendor pages).
    @State private var voice = LocalMicVoiceSource()

    var body: some View {
        // The view binds the source; it never connects it.
        SinuaOrb(pattern: .speaking, voice: voice)
            .frame(width: 160, height: 160)
            .task {
                // Needs NSMicrophoneUsageDescription in Info.plist; the system asks once.
                do { try await voice.connect() } catch { print("voice failed to start:", error) }
            }
            .onDisappear { voice.disconnect() }
    }
}

File: apps/site/snippets/voice/overview.kt
package snippets.voice

import androidx.compose.foundation.layout.size
import androidx.compose.runtime.Composable
import androidx.compose.runtime.DisposableEffect
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import dev.sinua.view.generated.SinuaOrb
import dev.sinua.view.generated.SinuaOrbPattern
import dev.sinua.voice.LocalMicVoiceSource

@Composable
fun TalkingOrb(micGranted: Boolean) {   // RECORD_AUDIO, requested by your app first
    // Any VoiceSource: the mic here, or a vendor source (see the vendor pages).
    val voice = remember { LocalMicVoiceSource() }
    DisposableEffect(micGranted) {
        if (micGranted) voice.connect()   // the view binds the source; it never connects it
        onDispose { voice.disconnect() }
    }
    SinuaOrb(pattern = SinuaOrbPattern.SPEAKING, voice = voice, modifier = Modifier.size(160.dp))
}
